src/Common/tests/Tests/System/Net/VirtualNetworkStreamTest.cs

```diff
@@ -17,17 +17,25 @@ public class VirtualNetworkStreamTest
         public void VirtualNetworkStream_SingleThreadIntegrityTest_Ok()
         {
             var rnd = new Random();
+            var rndLock = new object();
+            
             var network = new VirtualNetwork();
 
             using (var client = new VirtualNetworkStream(network, isServer: false))
             using (var server = new VirtualNetworkStream(network, isServer: true))
             {
                 for (int i = 0; i < 100000; i++)
                 {
-                    int bufferSize = rnd.Next(1, 2048);
-
-                    byte[] writeFrame = new byte[bufferSize];
-                    rnd.NextBytes(writeFrame);
+                    int bufferSize;
+                    byte[] writeFrame;
+                    
+                    lock (rndLock)
```

Comment:
Why is this lock needed?  rnd isn't exposed to any other threads and is only used by this function locally.